core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -390,6 +392,14 @@ public void testFromXContent() throws IOException {
         }
     }
 
+    /**
+     * subclasses should override this method in case some fields in xContent should be protected from random
+     * shuffling in the {@link #testFromXContent()} test case
+     */
+    protected Set<String> provideShuffleproofFields() {
```

Comment:
I expected we need to override this method for other cases. like I have memories of indices query depending on the order of keys. or did we fix that as part of the query refactoring?